kube/src/api/metadata.rs

```diff
@@ -2,84 +2,9 @@ use k8s_openapi::apimachinery::pkg::apis::meta::v1::OwnerReference;
-///
-/// And these optional properties:
-/// - `.metadata.namespace`
-/// - `.metadata.resource_version`
-///
-/// This avoids a bunch of the unnecessary unwrap mechanics for apps.
-pub trait Resource {
-    /// Type information for types that do not know their resource information at compile time.
-    ///
-    /// Types that know their metadata at compile time should select `DynamicType = ()`.
-    /// Types that require some information at runtime should select `DynamicType`
-    /// as type of this information.
-    ///
-    /// See [`DynamicObject`](crate::api::DynamicObject) for a valid implementation of non-k8s-openapi resources.
-    type DynamicType: Send + Sync + 'static;
-
-    /// Returns kind of this object
-    fn kind(dt: &Self::DynamicType) -> Cow<'_, str>;
-    /// Returns group of this object
-    fn group(dt: &Self::DynamicType) -> Cow<'_, str>;
-    /// Returns version of this object
-    fn version(dt: &Self::DynamicType) -> Cow<'_, str>;
-    /// Returns apiVersion of this object
-    fn api_version(dt: &Self::DynamicType) -> Cow<'_, str> {
-        let group = Self::group(dt);
-        if group.is_empty() {
-            return Self::version(dt);
-        }
-        let mut group = group.into_owned();
-        group.push('/');
-        group.push_str(&Self::version(dt));
-        group.into()
-    }
-    /// Returns the plural name of the kind
-    ///
-    /// This is known as the resource in apimachinery, we rename it for disambiguation.
-    /// By default, we infer this name through pluralization.
-    ///
-    /// The pluralization process is not recommended to be relied upon, and is only used for
-    /// `k8s_openapi` types, where we maintain a list of special pluralisations for compatibility.
-    ///
-    /// Thus when used with `DynamicObject` or `kube-derive`, we override this with correct values.
-    fn plural(dt: &Self::DynamicType) -> Cow<'_, str> {
-        to_plural(&Self::kind(dt).to_ascii_lowercase()).into()
-    }
-
-    /// Creates a url path for http requests for this resource
-    fn url_path(dt: &Self::DynamicType, namespace: Option<&str>) -> String {
-        let n = if let Some(ns) = namespace {
-            format!("namespaces/{}/", ns)
-        } else {
-            "".into()
-        };
-        let group = Self::group(dt);
-        let api_version = Self::api_version(dt);
-        let plural = Self::plural(dt);
-        format!(
-            "/{group}/{api_version}/{namespaces}{plural}",
-            group = if group.is_empty() { "api" } else { "apis" },
-            api_version = api_version,
-            namespaces = n,
-            plural = plural
-        )
-    }
-
-    /// Metadata that all persisted resources must have
-    fn meta(&self) -> &ObjectMeta;
-    /// Metadata that all persisted resources must have
-    fn meta_mut(&mut self) -> &mut ObjectMeta;
-}
+pub use kube_core::resource::Resource;
 
 /// Helper methods for resources.
 pub trait ResourceExt: Resource {
```

Comment:
The ResourceExt trait should probably be there as well.

Reply:
Ok, wasn't sure about that.